addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoDeviceHandler.java

```diff
@@ -79,25 +80,30 @@ public void initialize() {
     }
 
     private void scheduleRefreshJob() {
-        logger.debug("Scheduling update channel thread in {} s", refreshStrategy.nextRunDelayInS());
+        long delay = refreshStrategy.nextRunDelayInS();
+        logger.debug("Scheduling update channel thread in {} s", delay);
         refreshJob = scheduler.schedule(() -> {
             updateChannels();
-            refreshJob.cancel(false);
-            refreshJob = null;
+            if (refreshJob != null && !refreshJob.isCancelled()) {
+                logger.debug("cancel refresh job");
+                refreshJob.cancel(false);
+                refreshJob = null;
+            }
             scheduleRefreshJob();
-        }, refreshStrategy.nextRunDelayInS(), TimeUnit.SECONDS);
+        }, delay, TimeUnit.SECONDS);
     }
 
     @Override
     public void dispose() {
         logger.debug("Running dispose()");
         if (refreshJob != null && !refreshJob.isCancelled()) {
+            logger.debug("cancel refresh job");
             refreshJob.cancel(true);
             refreshJob = null;
         }
     }
 
-    protected abstract DEVICE updateReadings();
+    protected abstract DEVICE updateReadings() throws RetrofitError;
```

Comment:
Its a `RuntimeException` mentioning it here gives a false feeling of information because any method can throw it.

Reply:
Fair enough.